Core/include/Acts/Utilities/Helpers.hpp

```diff
@@ -146,32 +147,34 @@ double eta(const Eigen::MatrixBase<Derived>& v) noexcept {
   return std::atanh(v[2] / v.norm());
 }
 
-/// Helper method to cast out the binning value from a 3D Vector
+/// Helper method to extract the binning value from a 3D vector.
 ///
 /// For this method a 3D vector is required to guarantee all potential
-/// binning values
-///
-static double cast(const Vector3D& position, BinningValue bval) {
-  if (bval < 3)
-    return position[bval];
+/// binning values.
+inline double cast(const Vector3D& position, BinningValue bval) {
   switch (bval) {
+    case binX:
+      return position[0];
+    case binY:
+      return position[1];
+    case binZ:
+      return position[2];
     case binR:
       return perp(position);
-      break;
     case binPhi:
       return phi(position);
-      break;
+    case binRPhi:
+      return perp(position) * phi(position);
     case binH:
       return theta(position);
-      break;
     case binEta:
       return eta(position);
-      break;
     case binMag:
       return position.norm();
-      break;
+    default:
+      assert("Invalid BinningValue enum value");
```

Comment:
I think this never raises an error.

```suggestion
      assert(false, "Invalid BinningValue enum value");
```
?

Reply:
Fixed.